go/vt/vtgate/engine/route.go

```diff
@@ -571,6 +623,32 @@ func (route *Route) resolveSingleShard(vcursor VCursor, bindVars map[string]*que
 	return newKeyspace, shard, ksid, nil
 }
 
+func (route *Route) updateChangedVindexes(subQueryResult *sqltypes.Result, vcursor VCursor, bindVars map[string]*querypb.BindVariable, ksid []byte) error {
+	if len(route.ChangedVindexValues) == 0 {
+		return nil
+	}
+	for i, colVindex := range route.Table.Owned {
+		if colValue, ok := route.ChangedVindexValues[colVindex.Name]; ok {
+			resolvedVal, err := colValue.ResolveValue(bindVars)
+			if err != nil {
+				return err
+			}
+			ids := make([]sqltypes.Value, 0, len(subQueryResult.Rows))
+
+			for _, row := range subQueryResult.Rows {
```

Comment:
After some more thinking, I feel like we should return an error if number of rows is >1.

This is because we are essentially admitting in the next statement that multiple rows own a vindex entry, which contradicts the concept of the owned vindex.

We can later look at relaxing this rule if someone comes up with a legitimate use case. Until then, let's play it safe.